src/main/java/org/folio/circulation/domain/Loan.java

```diff
@@ -222,35 +222,43 @@ Loan withProxy(User newProxy) {
     return new Loan(representation, item, user, newProxy, checkinServicePoint,
       checkoutServicePoint, originalDueDate);
   }
-  
+
   public Loan withCheckinServicePoint(ServicePoint newCheckinServicePoint) {
     return new Loan(representation, item, user, proxy, newCheckinServicePoint,
       checkoutServicePoint, originalDueDate);
   }
-  
+
   public Loan withCheckoutServicePoint(ServicePoint newCheckoutServicePoint) {
     return new Loan(representation, item, user, proxy, checkinServicePoint,
       newCheckoutServicePoint, originalDueDate);
   }
 
-  private void changeLoanPolicy(String newLoanPolicyId) {
+  private void changeLoanPolicyId(String newLoanPolicyId) {
     if (newLoanPolicyId != null) {
       representation.put("loanPolicyId", newLoanPolicyId);
     }
   }
-  
+
+  private void changeLoanPolicyName(String newLoanPolicyName) {
+    if (Objects.nonNull(newLoanPolicyName) ) {
+      representation.put(LoanProperties.LOAN_POLICY_NAME, newLoanPolicyName);
+    }
+  }
+
   public ServicePoint getCheckinServicePoint() {
     return this.checkinServicePoint;
   }
-  
+
   public ServicePoint getCheckoutServicePoint() {
     return this.checkoutServicePoint;
   }
 
-  public Loan renew(DateTime dueDate, String basedUponLoanPolicyId) {
+  public Loan renew(DateTime dueDate, String basedUponLoanPolicyId,
```

Comment:
Eventually `basedUponLoanPolicyId` and `basedUponLoanPolicyName` should really be replaced by an object rather than two raw properties.

Reply:
I agree that it should be done eventually. Do you think we need a separate ticket for that?

Reply:
I was mainly referring to the representation.

Reply:
There are lots of improvements needed in this area. I haven't raised tickets for each because they tend to go stale quickly.

In this case, the loan policy itself could be passed instead of the two properties.